src/category_theory/sigma/basic.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.natural_isomorphism
+import category_theory.eq_to_hom
+import data.sigma.basic
+import category_theory.pi.basic
+
+/-!
+# Disjoint union of categories
+
+We define the category structure on a sigma-type (disjoint union) of categories.
+-/
+
+namespace category_theory
+namespace sigma
+
+universes w₁ w₂ w₃ v₁ v₂ u₁ u₂
+
+variables {I : Type w₁} {C : I → Type u₁} [Π i, category.{v₁} (C i)]
+
+/--
+The type of morphisms of a disjoint union of categories: for `X : C i` and `Y : C j`, a morphism
+`(i, X) ⟶ (j, Y)` if `i = j` is just a morphism `X ⟶ Y`, and if `i ≠ j` there are no such morphisms.
+-/
+inductive sigma_hom : (Σ i, C i) → (Σ i, C i) → Type (max w₁ v₁ u₁)
+| mk : Π {i : I} {X Y : C i}, (X ⟶ Y) → sigma_hom ⟨i, X⟩ ⟨i, Y⟩
+
+namespace sigma_hom
+
+/-- The identity morphism on an object. -/
+def id : Π (X : Σ i, C i), sigma_hom X X
+| ⟨i, X⟩ := mk (𝟙 _)
+
+instance (X : Σ i, C i) : inhabited (sigma_hom X X) := ⟨id X⟩
+
+/-- Composition of sigma homomorphisms. -/
+def comp : Π {X Y Z : Σ i, C i}, sigma_hom X Y → sigma_hom Y Z → sigma_hom X Z
+| _ _ _ (mk f) (mk g) := mk (f ≫ g)
```

Comment:
Here and in four other places in this PR, can you get rid of all those `_`s by moving various things before the colon.
```suggestion
def comp {X Y Z : Σ i, C i} : sigma_hom X Y → sigma_hom Y Z → sigma_hom X Z
| (mk f) (mk g) := mk (f ≫ g)
```
(untested, sorry! Am in the middle of editing another branch of mathlib. Am I wrong?)

Reply:
Unfortunately the story here is a little more subtle - putting the things before the colon means they're fixed in the equation compiler, but the presence of the constructor `mk` actually ensures that `X` is of the form `i, X'`, and similarly for Y and Z, and it additionally ensures that all of the `i`s are the same, so it makes a difference whether they're before or after the colon. 
Similarly for the others below